dev/src/pool.ts

```diff
@@ -42,6 +42,12 @@ export class ClientPool<T> {
    */
   private terminated = false;
 
+  /**
+   * Deferred promise that is resolved when there are no active operations on
+   * the client pool.
+   */
+  private operationsDeferred = new Deferred<void>();
```

Comment:
Can you come up with a more descriptive name?

Reply:
there are no names I could come up with that sound good: operationsFinishedDeferred, operationsCompletedDeferred, allRequestsCompleted, ready4terminatedeferred